lib/src/command/deps.dart

```diff
@@ -4,12 +4,37 @@
 
 import 'dart:collection';
 
+import 'package:analyzer/analyzer.dart' as analyzer;
+import 'package:path/path.dart' as p;
+
 import '../ascii_tree.dart' as tree;
 import '../command.dart';
+import '../dart.dart';
 import '../log.dart' as log;
 import '../package.dart';
+import '../package_name.dart';
 import '../utils.dart';
 
+/// Returns `true` if [path] looks like a Dart entrypoint.
+bool _isDartExecutable(String path) {
+  if (p.extension(path) == '.dart') {
+    try {
+      var unit = analyzer.parseDartFile(path, parseFunctionBodies: false);
+      return isEntrypoint(unit);
+    } on analyzer.AnalyzerErrorGroup {
+      return false;
+    }
+  }
+
+  return false;
+}
+
+/// The immediate non-dev dependencies this package specifies in its pubspec.
+List<PackageRange> _nonDevDependencies(Package package) {
```

Comment:
It does not include `dependencyOverrides` or I should not care about them in such a case? (As they have the same binaries names)